System/OsPath/Common.hs

```diff
@@ -45,11 +45,13 @@ module System.OsPath
   , PS.unsafeEncodeUtf
   , PS.encodeWith
   , encodeFS
+#if !defined(__MHS__)
```

Comment:
I think it's more idiomatic to write `#ifdef MIN_VERSION_template_haskell`: we don't really care MHS or not, we care if it has TH (and, in glorious future, more specifically [`template-haskell-quasiquoter`](https://hackage.haskell.org/package/template-haskell-quasiquoter-0.1.0.0)).

Reply:
Yes, that is a better way.

Reply:
I've made some updates to minimize the changes.